R/core_serialize.R

```diff
@@ -47,7 +47,7 @@ writeObject <- function(con, object, writeType = TRUE) {
   type <- class(object)[[1]]
 
   if (type %in% c("integer", "character", "logical", "double", "numeric", "factor", "Date", "POSIXct")) {
-    if (is.na(object)) {
+    if (is.na(object) && !is.nan(object)) {
```

Comment:
TODO: create at least 1 test case for this. For example:

```
library(sparklyr)

sc <- spark_connect(master = "local")
print(invoke_static(sc, "sparklyr.Test", "infer", NaN))
```

Previous output: `"String"` (incorrect)
Expected output (which was indeed the case after this change): `"Double"`